gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -178,8 +179,10 @@ private void showAddFeatureDialog(Pair<ImmutableList<Layer>, Point> args) {
           if (layer.getContributorsCanAdd().contains(FeatureType.POINT)
               && layer.getContributorsCanAdd().contains(FeatureType.POLYGON)) {
             showFeatureTypeDialog(layer, point);
-          } else {
+          } else if (layer.getContributorsCanAdd().contains(FeatureType.POINT)) {
```

Comment:
Nit: A more cleaner way to do this might be to refactor this logic into its own method, and do something like::
```java
if (layer.getContributorsCanAdd().isEmpty()) {
  Timber.e("No permissions set on layer %s%, layer.getId());
}

if (layer.size() > 1) {
  showFeatureTypeDialog(layer, point);
  return;
}

FeatureType allowedType = layer.getContributorsCanAdd().get(0);
if (allowedType == FeatureType.POINT) {
  viewModel.addFeature(layer, point);
} else if (allowedType ==  FeatureType.POLYGON} {
  showPolygonInfoDialog(layer);
} else {
  Timber.w("Unsupported feature type defined in layer: %s", allowedType);
}
```

Reply:
updated.